packages/js/settings-editor/src/legacy/form.tsx

```diff
@@ -50,30 +60,60 @@ export const Form = ( {
 		return formData;
 	};
 
-	const handleSubmit = ( event: React.FormEvent< HTMLFormElement > ) => {
+	const handleSubmit = async (
+		event: React.FormEvent< HTMLFormElement >
+	) => {
 		event.preventDefault();
+		setIsBusy( true );
 
 		const query: Record< string, string > = {
 			page: 'wc-settings',
-			tab: settingsPage.slug,
 		};
+		if ( settingsPage.slug !== 'general' ) {
+			query.tab = settingsPage.slug;
+		}
 		if ( activeSection !== 'default' ) {
 			query.section = activeSection;
 		}
 
-		const formData = getFormData();
-		formData.save = 'Save changes';
-		formData._wpnonce = settingsData._wpnonce;
-		formData._w_http_referer = '/wp-admin/' + getNewPath( query );
-
-		// eslint-disable-next-line no-console
-		console.log(
-			'tab: ',
-			settingsPage.slug,
-			'section: ',
-			activeSection,
-			formData
+		const updatedData = getFormData();
+		updatedData.save = 'Save changes';
+		updatedData._wpnonce = settingsData._wpnonce;
+		updatedData._w_http_referer = '/wp-admin/' + getNewPath( query );
+
+		const payload = new FormData();
+		for ( const [ key, value ] of Object.entries( updatedData ) ) {
+			payload.append( key, value );
+		}
+
+		const response = await fetch(
+			`/?rest_route=/wc-admin/legacy-settings&${ new URLSearchParams(
```

Comment:
This might cause an issue if the store is in a subdirectory. For example, if the URL is `https://example.com/subdirectory/`, the fetch call to `/?rest_route=/wc-admin/legacy-settings` will return an invalid response. I wonder if we can use `apiFetch` to make the request. I think it will handle this automatically.

Reply:
Nice, fixed in [use apiFetch for api call](https://github.com/woocommerce/woocommerce/pull/55467/commits/812f9643b150d61db1d1b91259156b640c637a60)